src/metaxy/metadata_store/postgres.py

```diff
@@ -0,0 +1,1158 @@
+            # Table exists, insert data
+            # Serialize struct columns for insert with JSONB cast
+            if self._struct_compat_mode and has_struct_columns:
+                df_serialized = self._serialize_struct_columns(df)
+                self._insert_with_jsonb_cast(table_name, df_serialized)
+            else:
+                conn.insert(table_name, obj=df)
+
+    def read_metadata_in_store(
+        self,
+        feature: FeatureKey | type[BaseFeature],
+        *,
+        feature_version: str | None = None,
+        filters: Sequence[nw.Expr] | None = None,
+        columns: Sequence[str] | None = None,
+        **kwargs: Any,
+    ) -> nw.LazyFrame[Any] | None:
+        """Read metadata from store.
+
+        In struct compatibility mode with JSONB storage, Ibis reads JSONB natively.
+        Deserialization to Polars structs happens at the final boundary when returning to users.
+        """
+        feature_key = self._resolve_feature_key(feature)
+        table_name = self.get_table_name(feature_key)
+        if table_name not in self.conn.list_tables():
+            return None
+
+        # Return Ibis LazyFrame directly
+        # JSONB columns are read natively by Ibis and can be queried with [] operator
+        lazy_frame = super().read_metadata_in_store(
+            feature, feature_version=feature_version, filters=filters, columns=columns
+        )
+        return lazy_frame
+
+    def _drop_feature_metadata_impl(self, feature_key: FeatureKey) -> None:
+        """Drop the table for a feature.
+
+        Args:
+            feature_key: Feature key to drop metadata for
+        """
+        table_name = self.get_table_name(feature_key)
+        if table_name in self.conn.list_tables():
+            self.conn.drop_table(table_name)
+
+    def display(self) -> str:
+        """Display string for this store."""
+        details: list[str] = []
+        if self.database:
+            details.append(f"database={self.database}")
+        if self.schema:
+            details.append(f"schema={self.schema}")
+        if self.host:
+            details.append(f"host={self.host}")
+        if self.port:
+            details.append(f"port={self.port}")
+
+        detail_str = ", ".join(details)
+        if detail_str:
+            return f"PostgresMetadataStore({detail_str})"
+        if self.connection_string:
+            sanitized = sanitize_uri(self.connection_string)
+            return f"PostgresMetadataStore(connection_string={sanitized})"
+        return "PostgresMetadataStore()"
+
+    def _has_native_struct_support(self) -> bool:
+        """
+        NOTE: Native struct support for PostgreSQL in the Ibis/sqlglot stack has proven
+        unreliable across different environments. We are temporarily disabling it
+        to enforce the more stable JSON serialization path.
+        """
+        return False
+
+    @staticmethod
+    def _serialize_struct_column(df: pl.DataFrame, column_name: str) -> pl.DataFrame:
```

Comment:
Looks like this method should belong somewhere upstream, it's not specific to Postgresql.